main/php_ini.c

```diff
@@ -410,6 +410,66 @@ static void php_load_zend_extension_cb(void *arg) { }
 #endif
 /* }}} */
 
+/* {{{ get_env_location
+ */
+static char* get_env_location(const char *envname)
+{
+	char *env_location = getenv(envname);
+#ifdef PHP_WIN32
+	char phprc_path[MAXPATHLEN];
+#endif
+
+#ifdef PHP_WIN32
+	if (!env_location) {
+		char dummybuf;
+		int size;
+
+		SetLastError(0);
+
+		/*If the given buffer is not large enough to hold the data, the return value is
+		the buffer size,  in characters, required to hold the string and its terminating
+		null character. We use this return value to alloc the final buffer. */
+		size = GetEnvironmentVariableA(envname, &dummybuf, 0);
+		if (GetLastError() == ERROR_ENVVAR_NOT_FOUND) {
+			/* The environment variable doesn't exist. */
+			return "";
+		}
+
+		if (size == 0) {
+			return "";
+		}
+
+		size = GetEnvironmentVariableA(envname, phprc_path, size);
+		if (size == 0) {
+			return "";
+		}
+
+		env_location = phprc_path;
```

Comment:
You are returning a reference to stack memory here. The pointer will be invalidated when the function returns.

Reply:
suggestions? always strdup value, return NULL if no value? playing with is_allocated variable seems overkill, as this code path is invoked only on init (not request init)?

Reply:
One option would be to not return anything from `get_env_location`, but to pass the buffer into the function.